src/ProjectTemplates/ComponentsWebAssembly.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Program.cs

```diff
@@ -19,7 +21,17 @@ public static async Task Main(string[] args)
             var builder = WebAssemblyHostBuilder.CreateDefault(args);
             builder.RootComponents.Add<App>("app");
 
+#if (!Hosted || NoAuth)
             builder.Services.AddSingleton(new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
+#else
+            builder.Services.AddScoped(sp =>
+            {
+                var handler = sp.GetRequiredService<BaseAddressAuthorizationMessageHandler>();
+                handler.InnerHandler = new HttpClientHandler();
```

Comment:
It was not possible to set the handler in the constructor otherwise HttpClientFactory would barf about it on the tests. If we want to "hide" the boilerplate we can add a method to the handler that sets it up and we would simply call that on startup.

Reply:
Not setting the handler here doesn't work either. So we either leave it like this (not terrible, move it to use HttpClientFactory, which does the right thing, or add a method that can set the client and returns the handler itself (so that it is an expression and we can make this a lambda))

Reply:
What are the tests that have problems?

I wouldn’t normally expect us to make API decisions based on limitations of our own test code, if that’s what you mean. Or is there some way that setting it would break user code?

Reply:
Sorry if I was not clear. Setting the hanlder breaks HttpClientFactory scenarios (which our tests use for coverage)

Reply:
At this point I think I'm ready to move the needle on this and just use HttpClientFactory on the templates too, as it seems simpler.

Reply:
HttpClientFactory sounds fine to me if you think it's a preferable pattern for customers.